### PR TITLE
aggregated benchmark results

### PR DESCRIPTION
**Pull request by JudeNiroshan**
### Description

Added a new documentation file that contains summarized benchmark results.

### CODE REVIEW

File: benchmark_results.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+#### VM Details: 🖥️
+TBD
+
+#### Config Details: ⚒️
+
+- Java version: openjdk 11.0.12 2021-07-20 LTS
+- Persistence: Mongo DB v4.4.9
+- Kogito version: 1.11.0.Final
+- Concurrence Strategy: CONST_CONCURRENT_USERS
```
Comment (dmartinol) [inline benchmark_results.md:18]: This applies to Gatling only. worth to mention it

File: benchmark_results.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+|  Gatling 	|  emptyProcess ➙ default |  ❌ No persistence	|  1 	|  60 	|   5 min	|  334250 	|  147883/186367 	|  28 ms 	|  1110.46/sec 	|
+|  Gatling 	|  notPersistedProcess ➙ default |  ❌ No persistence	|  1 	|  1 	|   5 min	|  150819 	|  131159/19660 	|  2 ms 	|  501.06/sec 	|
+|  Gatling 	|  notPersistedProcess ➙ default |  ❌ No persistence	|  1 	|  4 	|   5 min	|  173586 	|  144145/29441 	|  2 ms 	|  576.69/sec 	|
+|  Gatling 	|  notPersistedProcess ➙ default |  ❌ No persistence	|  1 	|  8 	|   5 min	|  178186 	|  146601/31585 	|  3 ms 	|  591.98/sec 	|
+|  Gatling 	|  notPersistedProcess ➙ default |  ❌ No persistence	|  1 	|  20 	|   5 min	|  181766 	|  141168/40598 	|  12 ms 	|  603.87/sec 	|
+|  Gatling 	|  notPersistedProcess ➙ default |  ❌ No persistence	|  1 	|  40 	|   5 min	|  187024 	|  153349/33675 	|  27 ms 	|  621.34/sec 	|
+|  Gatling 	|  notPersistedProcess ➙ default |  ❌ No persistence	|  1 	|  60 	|   5 min	|  194124 	|  150145/43979 	|  43 ms 	|  644.93/sec 	|
+|  Java-Quarkus 	|  simple ➙ default |  ✅ Mongo DB ➙ clean-before	|  1 	|  1 	|   2 min	|  10777 	|  10777/0 	|  10 ms 	|  89/sec 	|
```
Comment (dmartinol) [inline benchmark_results.md:36]: Can you add quick description of what "clean-before" and "id-index" mean, from a procedure perspective?

File: benchmark_results.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+TBD
+
+#### Config Details: ⚒️
+
+- Java version: openjdk 11.0.12 2021-07-20 LTS
+- Persistence: Mongo DB v4.4.9
+- Kogito version: 1.11.0.Final
+- Concurrence Strategy: CONST_CONCURRENT_USERS
+
+## VM deployment
```
Comment (dmartinol) [inline benchmark_results.md:20]: Shouldn't we split the report in 2 sections, one for the default configuration (e.g., to be used for official report) and one with the manual updates to boost the performance? Or do we want Ivo to make the split? (leaving the team to decide)

File: benchmark_results.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+|   Java-Quarkus	|  simple ➙ pool&workers=100 |  ✅ Mongo DB ➙ id-index	|  2 	|  4 	|   2 min	|  29132 	|  29132/0 	|  15 ms 	|   242/sec	|
+|   Java-Quarkus	|  simple ➙ pool&workers=100 |  ✅ Mongo DB ➙ id-index	|  5 	|  4 	|   2 min	|  20958 	|  20958/0 	|  22 ms 	|   174/sec	|
+|   Java-Quarkus	|  simple ➙ pool&workers=100 |  ✅ Mongo DB ➙ id-index	|  1 	|  8 	|   2 min	|  59936 	|  59936/0 	|  15 ms 	|   499/sec	|
+|   Java-Quarkus	|  simple ➙ pool&workers=100 |  ✅ Mongo DB ➙ id-index	|  2 	|  8 	|   2 min	|  46515 	|  46515/0 	|  20 ms 	|   387/sec	|
+|   Java-Quarkus	|  simple ➙ pool&workers=100 |  ✅ Mongo DB ➙ id-index	|  5 	|  8 	|   2 min	|  26142 	|  26142/0 	|  36 ms 	|   217/sec	|
+|   Java-Quarkus	|  simple ➙ pool&workers=100 |  ✅ Mongo DB ➙ id-index	|  1 	|  100 	|   2 min	|  117212 	|  117212/0 	|  102 ms 	|   971/sec	|
+|   Java-Quarkus	|  simple ➙ pool&workers=100 |  ✅ Mongo DB ➙ id-index	|  2 	|  100 	|   2 min	|  118294 	|  118294/0 	|  100 ms 	|   985/sec	|
+|   Java-Quarkus	|  simple ➙ pool&workers=100 |  ✅ Mongo DB ➙ id-index	|  5 	|  100 	|   2 min	|  47730 	|  47730/0 	|  250 ms 	|   397/sec	|
```
Comment (dmartinol) [inline benchmark_results.md:96]: Shouldn't we also add the csv file with the same results that can be downloaded to easily filter and extract data? I'm not sure we can do the same data manipulation from this Markdown table

**Review by dmartinol — COMMENTED** [review]
Please consider the inline comments, good job for now

File: benchmark_results.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+# Benchmark Results
+
+Following numbers were gathered targeting the [process-quarkus-example](https://github.com/kiegroup/kogito-examples/tree/stable/process-quarkus-example) on different compute environments.
```
[the remark this answers is not in the document]
Comment (JudeNiroshan) [reply benchmark_results.md]: 4e2b57d885604a2d143d5af939ec37d92383a62c

File: benchmark_results.md (hunk 3 above)
Comment (dmartinol) [reply benchmark_results.md:32]: keep as it is, single column is also fine

**Pull request merged**